letsencrypt/cli.py

```diff
@@ -896,14 +896,12 @@ def _paths_parser(helpful):
     cph = "Path to where cert is saved (with auth), installed (with install --csr) or revoked."
     if verb == "auth":
         add("paths", "--cert-path", default=flag_default("auth_cert_path"), help=cph)
-    elif verb == "revoke":
-        add("paths", "--cert-path", type=read_file, required=True, help=cph)
     else:
-        add("paths", "--cert-path", help=cph, required=(verb == "install"))
+        add("paths", "--cert-path", help=cph,
+            required=(verb in ["install", "revoke"]))
 
     # revoke --key-path reads a file, install --key-path takes a string
-    add("paths", "--key-path", type=((verb == "revoke" and read_file) or str),
-        required=(verb == "install"),
+    add("paths", "--key-path", required=(verb == "install"),
```

Comment:
@pde `X and Y or Z` trick should not be used in Python: we have `Y if X else Z` if you really need it